packages/apollo/src/commands/service/check.ts

```diff
@@ -285,7 +288,13 @@ export default class ServiceCheck extends ProjectCommand {
                 task.output = "Resolving schema";
                 taskOutput.shouldOutputJson = flags.json;
 
-                if (flags.federated) {
+                // If we're `federated`, then run composition validation. When we're using composition
+                // validation we'll receive a schema has that represents the composed schema.
+                if (serviceName) {
```

Comment:
I think removing `flags.federated` here is a mistake, unless we're trying to remove that flag. I'll point back to the comment about this possibly being a bit too implicit. Does the presence of `serviceName` -> we're federated?

Reply:
Matt explicitly said he wants to use `--serviceName` and not `--federated`. Yes, `serviceName` means we're federated.

Reply:
Ok. We should document this in the `serviceName` flag `description` up above.

We should also remove the `federated` flag if it does nothing now. It's _sort of_ a breaking change, but since `serviceName` has been a requirement as well, nothing's actually breaking, we just don't _need_ it anymore.

Reply:
I want to point out that this is a breaking change, but the current state of the CLI doesn't allow anyone to run federated service checks at all, so it's not going to break anyone :)